src/java.base/share/classes/java/lang/StringCoding.java

```diff
@@ -25,15 +26,46 @@
 
 package java.lang;
 
+import jdk.internal.misc.Unsafe;
 import jdk.internal.vm.annotation.IntrinsicCandidate;
 
+import static jdk.internal.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+
 /**
  * Utility class for string encoding and decoding.
  */
 class StringCoding {
+    private static final Unsafe UNSAFE = Unsafe.getUnsafe();
+    private static final long POSITIVE_MASK = 0b1000000_1000000_1000000_1000000_1000000_1000000_1000000_1000000L;
```

Comment:
Shouldn't this be `0b10000000_10000000_10000000_10000000_10000000_10000000_10000000_10000000L`? Either way this new method needs a sanity test.